ext/date/php_date.c

```diff
@@ -2500,6 +2500,45 @@ PHPAPI bool php_date_initialize(php_date_obj *dateobj, const char *time_str, siz
 	return 1;
 } /* }}} */
 
+PHPAPI void php_date_initialize_from_ts_long(php_date_obj *dateobj, zend_long sec, int usec) /* {{{ */
+{
+	dateobj->time = timelib_time_ctor();
+	dateobj->time->zone_type = TIMELIB_ZONETYPE_OFFSET;
+
+	timelib_unixtime2gmt(dateobj->time, (timelib_sll)sec);
+	timelib_update_ts(dateobj->time, NULL);
+	php_date_set_time_fraction(dateobj->time, usec);
+} /* }}} */
+
+PHPAPI bool php_date_initialize_from_ts_double(php_date_obj *dateobj, double ts) /* {{{ */
+{
+	double sec_dval = trunc(ts);
+	zend_long sec;
+	int usec;
+
+	if (UNEXPECTED(isnan(sec_dval)
+		|| sec_dval >= (double)TIMELIB_LONG_MAX
+		|| sec_dval < (double)TIMELIB_LONG_MIN
+	)) {
+		zend_throw_error(date_ce_date_range_error,
+						"Seconds must be a finite number between "TIMELIB_LONG_FMT" and "TIMELIB_LONG_FMT", %g given",
```

Comment:
Could you add a space around the two `TIMELIB_LONG_*` constant names here? i had to do a double take to read this.